src/rp2_common/hardware_dma/include/hardware/dma.h

```diff
@@ -126,7 +126,7 @@ typedef struct {
  *             Usually disabled for peripheral to memory transfers
  */
 static inline void channel_config_set_read_increment(dma_channel_config *c, bool incr) {
-    c->ctrl = incr ? (c->ctrl | DMA_CH0_CTRL_TRIG_INCR_READ_BITS) : (c->ctrl & ~DMA_CH0_CTRL_TRIG_INCR_READ_BITS);
+  c->ctrl = (c->ctrl & ~DMA_CH0_CTRL_TRIG_INCR_READ_BITS) | (incr * DMA_CH0_CTRL_TRIG_INCR_READ_BITS);
```

Comment:
@kilograham Any idea why this is doing RMW rather than using `hw_set_bits` / `hw_clear_bits` ?

Reply:
because it isn't a hardware register.. also this optimization is unnecessary; the function call is nearly always inlined with a constant incr, and even if not the compiler does just a good a job with the code as is.

Reply:
Right, I forgot conditional execution of ARM